### ISSUE TITLE
BertForSequenceClassification defines criterion at every forward pass

### ISSUE THREAD

**Issue by KrishPro** [huggingface/transformers#15745]
## Environment info
I don't think, That any environmental info will be helpful in my issue

### Who can help

@LysandreJik, Can you look into the code ?

I just want anyone to have a look here https://github.com/huggingface/transformers/blob/master/src/transformers/models/bert/modeling_bert.py#L1579
`loss_fct = CrossEntropyLoss()`

Inside the forward function of `BertForSequenceClassification`, Instance of `CrossEntropyLoss` is created.
Which directly means that at every forward pass to the model we will create a instance of `CrossEntropyLoss`

Isn't this a performance bug ?

**Comment by KrishPro** [comment]
I'm planning to post a pull-request, If someone from the official contributors confirms that this is a bug
Because I'm still not sure if this is a bug or my understanding is un-clear

**Comment by KrishPro** [comment]
Ok, I myself found a quick-fix for this

If we don't want to create `CrossEntropyLoss`'s instance at every forward step, Then we can avoid passing labels to the forward functions and then calculate the loss as usual from `logits` from the model's output
